src/lib/support/BytesData.h

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <cstdlib>
+
+namespace chip {
+
+/**
+ * @brief A wrapper class for holding bytes buffer and its length, without the ownership of buffer.
+ */
+class BytesData
```

Comment:
nit: I wonder if an alternative name could be found. - I have not seen 'BytesData' used before for such a thing, it is often 'Span' (https://en.cppreference.com/w/cpp/container/span) or just Buffer/SizedBuffer may also be used. Data tends to make me think at something that owns the buffer (data owns, span/reference/pointer does not)

Reply:
Pigweed seems to use directly `using ByteSpan = std::span<std::byte>;` although I am surprised it can since span seems to be a C++20 construct.

Reply:
As an alternative, we could grab https://github.com/gsl-lite/gsl-lite in third_party and use the <span> there until we have C++20 support. That library also seems to bring in things that seem useful like pointer ownership, null/not-null markers and finalization support.

Reply:
Just learned that C++20 has std::span, which is what exactly we want. In this case, I renamed it to tamplate<T>Span, and `using ByteSpan = Span<uint8_t>`. Also renamed the methods to size() and data(), which is align with the C++ spec, I think this makes it easier if we move to std::span in the future.
As for the buffers that we holds ownership, I think we can use `PacketBufferHandle`?

Reply:
Note that our build environment is C++17 (I think) not 20. I was told pigweed provides a polyfill for it but there may be usage concerns @mspang 

Naming wise though, span seems consistent for future upgrades.

Reply:
After reviewing more we are actually just C++14 even. 
In any case , closing the comment. I am happy with the Span naming. For follow up PRs I believe we could add more features (std::span has quite a few).